PS-5408: Release notes for v5.7.25-28 (#2996)
percona/percona-server: doc/source/release-notes/Percona-Server-5.7.25-28.rst

.. rn:: 5.7.25-28

================================================================================
|Percona Server| |release|
================================================================================

Percona is glad to announce the release of |Percona Server| |release| on
|date|. Downloads are available `here
<http://www.percona.com/downloads/Percona-Server-5.7/Percona-Server-5.7.25-28/>`_
and from the :doc:`Percona Software Repositories </installation>`.
	
This release is based on `MySQL 5.7.25
<http://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-25.html>`_ and includes
all the bug fixes in it. |Percona Server| |release| is now the current GA
(Generally Available) release in the 5.7 series.
	
All software developed by Percona is open-source and free.
	
.. note:: 

   If you're currently using |Percona Server| 5.7, Percona recommends upgrading
   to this version of 5.7 prior to upgrading to |Percona Server| 8.0.

Bugs Fixed
================================================================================

> - FLUSH commands written to the binary log could cause errors in case

Reviewer: Must mention the new option, and PS-5405 must be completed too before the release

  of replication. Bug fixed :psbug:`1827`: (upstream :mysql:`88720`).

> - Running LOCK TABLES FOR BACKUP followed by STOP SLAVE SQL_THREAD

Reviewer: Upstream bug cannot be about LOCK TABLES FOR BACKUP, which is a Percona-only feature, please check

Author: This bug is mentioned in the jira ticket https://bugs.mysql.com/bug.php?id=93649
titled STOP SLAVE SQL_THREAD deadlocks if done while holding LOCK INSTANCE FOR BACKUP
Maybe using (related upstream ...) instead of (upstream ...) would make it better?

Reviewer: We fixed LOCK INSTANCE FOR BACKUP - which is an upstream command - and LOCK TABLES FOR BACKUP - which is only ours

Author: Ready for review

Reviewer: Upstream bug not mentioned

  could block replication preventing it from being restarted
  normally. Bug fixed :psbug:`4758` (upstream :mysqlbug:`93649`).
- InnoDB did not report the filename in I/O error messages. Bug fixed
  :psbug:`2455`.
- The ``ACCESS_DENIED`` field of the
  information_schema.user_statistics table was not updated
  correctly. Bug fixed :psbug:`3956`.
- MySQL could report that the maximum number of connections was

>   exceded with too many connections being in the CLOSE_WAIT state. Bug

Reviewer: ```suggestion
  exceeded with too many connections being in the CLOSE_WAIT state. Bug
```

  fixed :psbug:`4716` (upstream :mysqlbug:`92108`)
- ``http`` was replaced with ``https`` in http://bugs.percona.com in
  server crash messages. Bug fixed :psbug:`4855`.
- Wrong query results could be received in semi-join sub queries with
  materialization-scan that allowed inner tables of different
  semi-join nests to interleave. Bug fixed :psbug:`4907` (upstream bug
  :mysqlbug:`92809`).
- :psbug:`4911`: In some cases, the server using the the MyRocks
  storage engine could crash when TTL (Time to Live) was defined on a
  table.

> - In some ccases, running the SELECT stament with the ORDER BY and

Reviewer: ```suggestion
- In some cases, running the SELECT statement with the ORDER BY and
```

  LIMIT clauses could result in a less than optimal performance. Bug
  fixed :psbug:`4949` (ustream :mysql:`92850`)
- There was a typo in ``mysqld_safe.sh``: **trottling** was replaced
  with **throttling**. Bug fixed :psbug:`240`. Thanks to Michael
  Coburn for the patch.

Other bugs fixed:

> :psbug:`4705`,  

Reviewer: Check with @georgelorchpercona if this one needs a description

:psbug:`4791`, 

> :psbug:`4797`,

Reviewer: Check with @robgolebiowski if this one needs a description (it's a server not testsuite bug)

Reviewer: Does it?

Author: it doesn't

> :psbug:`4958`,

Reviewer: Check with @dutow if this one needs description (it's a server not testsuite bug)

Reviewer: Does it?

> :psbug:`4988`,

Reviewer: Check with @dutow if this one needs description

Reviewer: Does it?

> :psbug:`4989`,

Reviewer: Debug builds crashed if innodb_track_changed_pages variable was manipulated at runtime

Reviewer: Comment not addressed

> :psbug:`4996`,

Reviewer: Needs description

Author: Is it worth mentioning that it is related to https://jira.percona.com/browse/PS-3956 (access_denied component of userstats doesn't work)?

Reviewer: The bugs have been fixed together but I don't think they are related

Author: Ready for review

Reviewer: Still needs description

Author: Ready for review

:psbug:`5100`,

> :psbug:`5121`,

Reviewer: clang 7 compilation warnings have been fixed

> :psbug:`5158`,

Reviewer: Needs description, will be made public on release

> :psbug:`5219`,

Reviewer: Needs description

:psbug:`5268`,

> :psbug:`5366`

Reviewer: Needs description

<http://www.percona.com/downloads/Percona-Server-5.7/Percona-Server-5.7.25-28/>`_
and from the :doc:`Percona Software Repositories </installation>`.
	
This release is based on `MySQL 5.7.25
<http://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-25.html>`_ and includes
all the bug fixes in it. |Percona Server| |release| is now the current GA
(Generally Available) release in the 5.7 series.
	
All software developed by Percona is open-source and free.

In this release, |Percona Server| introduces the variable
:variable:`binlog_skip_flush_commands`. This variable controls whether
or not ``FLUSH`` commands are written to the binary log. Setting this
variable to **ON** can help avoid problems in replication. For more
information, see :ref:`percona-server.binary-log.flush.writing`.

.. note:: 

   If you're currently using |Percona Server| 5.7, Percona recommends upgrading
   to this version of 5.7 prior to upgrading to |Percona Server| 8.0.

Bugs Fixed
================================================================================

- FLUSH commands written to the binary log could cause errors in case
  of replication. Bug fixed :psbug:`1827`: (upstream :mysql:`88720`).

> - Running LOCK TABLES FOR BACKUP followed by STOP SLAVE SQL_THREAD

Reviewer: Why upstream bug reference was removed? It is a valid reference

  exceeded with too many connections being in the CLOSE_WAIT state. Bug
  fixed :psbug:`4716` (upstream :mysqlbug:`92108`)
- ``http`` was replaced with ``https`` in http://bugs.percona.com in
  server crash messages. Bug fixed :psbug:`4855`.
- Wrong query results could be received in semi-join sub queries with
  materialization-scan that allowed inner tables of different
  semi-join nests to interleave. Bug fixed :psbug:`4907` (upstream bug
  :mysqlbug:`92809`).
- :psbug:`4911`: In some cases, the server using the the MyRocks
  storage engine could crash when TTL (Time to Live) was defined on a
  table.
- In some cases, running the SELECT statement with the ORDER BY and
  LIMIT clauses could result in a less than optimal performance. Bug
  fixed :psbug:`4949` (ustream :mysql:`92850`)
- There was a typo in ``mysqld_safe.sh``: **trottling** was replaced
  with **throttling**. Bug fixed :psbug:`240`. Thanks to Michael
  Coburn for the patch.
- ``clang 7`` compilation warnings were fixed :psbug:`5121`.
- |MyRocks| could crash while running ``START TRANSACTION WITH
  CONSISTENT SNAPSHOT`` if other transactions were in specific
  states. Bug fixed :psbug:`4705`,

> - In some cases, ``mysqld` could crash when inserting data into a

Reviewer: ```suggestion
- In some cases, ``mysqld`` could crash when inserting data into a
```
